### PR TITLE
SQL Autocompletion

### LINKED ISSUE #195: autocomplete feature
Having an autocomplete feature will allow users to write queries faster. Jupyter already has a feature that autocompletes Python code so we can use that to guide this new feature.

The idea is to build a Jupyter extension that shows a tooltip with suggestions when users are editing cells that begin with %sql or %%sql, for example. if I have this:

```
%%sql
SELECT *
FROM TABLE
OR
```

And press tab, I should see a tooltip suggesting me to complete `ORDER BY`.

---

Docs for Jupyter extensions: https://jupyterlab.readthedocs.io/en/stable/user/extensions.html

In the past, we attempted to create an autocomplete extension for YAML files but we didn't finish it, however, the code might still be useful for this: https://github.com/ploomber/ploomber/pull/972/files

I think we can start with a very simple autocomplete that just suggests SQL keywords such as SELECT, FROM, ORDER BY, etc.

Once we have that working, we can make something more sophisticated that takes into account the SQL dialect, table, column names, etc.

### PR DESCRIPTION
**Pull request by neelasha23**
## Describe your changes
Added dependency for autocompletion extension.
Extension PR: https://github.com/ploomber/jupysql-autocompletion/pull/1/files

## Issue number

Closes #195 

## Checklist before requesting a review

- [x] Performed a self-review of my code
- [x] Formatted my code with [`pkgmt format`](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#lintingformatting)
- [x] Added [tests](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#testing) (when necessary).
- [x] Added documentation in the [docstring](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#documenting-changes-and-new-features) and [changelog](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--323.org.readthedocs.build/en/323/

### CODE REVIEW

**Review by tonykploomber — APPROVED** [review]
Adding dependency part LGTM,
@idomic Should we merge this PR UNTIL https://github.com/ploomber/jupysql-autocompletion/pull/1 got merged?

File: doc/howto/autocompletion.md [hunk 1]
```diff
@@ -32,6 +32,12 @@ conda create --name jupysqlenv python=3.10 jupyterlab --yes --channel conda-forg
 conda activate jupysqlenv
 ```
 
+This feature requires Node, if you don't have it, you can install it with conda:
+
+```sh
+conda install --channel conda-forge nodejs --yes
+```
```
Comment (edublancas) [inline doc/howto/autocompletion.md:39]: I believe node is only required for building the extension, right? It isn't needed for an end-user. If so, this should be part of the autocompletion package developer documentation

Comment (neelasha23) [reply doc/howto/autocompletion.md]: yeah you're right. Removed it from here.
It's mentioned in the [developer doc](https://github.com/ploomber/jupysql-autocompletion/blob/main/README.md)

**Comment by neelasha23** [comment]
Have resolved the comment. The CI docs build is failing right now with error `ERROR: No matching distribution found for jupysql-autocompletion`. Will rerun the jobs when the extension is uploaded to PyPI.
@edublancas

**Comment by edublancas** [comment]
ok, let's work on publishing the first version of the package so we can merge this

**Review by edublancas — APPROVED** [review]

**Pull request merged**